File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PurePursuit/Base/Coordination/Vector.java
package org.firstinspires.ftc.teamcode.PurePursuit.Base.Coordination;

public class Vector {

    public volatile double x, y;

    public Vector (double x, double y) {
        this.x = x;
        this.y = y;
    }

    public Vector (Vector vector) {
        this.x = vector.x;
        this.y = vector.y;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/math/Vector.kt
package org.firstinspires.ftc.teamcode.util.math

data class Vector(val mag: Double, val angle: Angle) {
    constructor(ijform: Point) : this(ijform.hypot, ijform.atan2)

    val ijform get() = Point(mag * angle.cos, mag * angle.sin)

    val norm: Vector get() = Vector(ijform / mag)

    operator fun plus(a: Vector) = Vector(a.ijform + ijform)
    operator fun minus(a: Vector) = Vector(a.ijform - ijform)
    operator fun times(scalar: Double) = Vector(mag * scalar, angle)
    operator fun div(scalar: Double) = this * (1 / scalar)
}
